Lemons/main.cpp

```diff
@@ -0,0 +1,97 @@
+//
+//  main.cpp
+//  lemonade
+//
+//  Created by Herman Sollerman on 27/03/16.
+//  Copyright © 2016 Herman Sollerman. All rights reserved.
+//
+
+#include <iostream>
+using namespace std;
+
+int main()
+{
+    int number_of_cases, number_of_days, lemon_per_cup, sugar_per_cup, cups_sold, lemon_price, total_cost=0, sugar_price, j, k;
+    FILE *input;
+
+    input = stdin;
+    fscanf(input, "%d", &number_of_cases);
+    
+    for (k=1; k <= number_of_cases; k++)
+    {
+        fscanf(input, "%d %d %d", &number_of_days, &lemon_per_cup, &sugar_per_cup);
+        int total_cups [number_of_days];
+        int lprices [number_of_days];
+        int sprices [number_of_days];
+        
+        int n=0;
+        
+        for (j=1; j<=number_of_days; j++)
+        {
+            fscanf(input, "%d %d %d", &cups_sold, &lemon_price, &sugar_price);
+            total_cups[n] = cups_sold;
+            lprices [n] = lemon_price;
+            sprices [n] = sugar_price;
+            ++n;
+        }
+        int cheap_lemon = lprices[0];
+        int lemons_to_buy = total_cups[0]*lemon_per_cup;
```

Comment:
Den här for-loopen skulle kunna bli lite lättare att förstå genom att man gör så här. Det är alltid bättre att ha så få specialfall som möjligt (kör loopen över alla dagar men sätt lemons_to_buy till noll till att börja med). Och det är oftast enklare att ha slutvillkoret efter loopen om man kan (istället för att kolla att man är på sista varvet).

 int lemons_to_buy = 0;
 for (int i=0; i<number_of_days; i++)
 {
     if (cheap_lemon<=lprices[i])
           lemons_to_buy += total_cups[i]*lemon_per_cup;
    else
    {
        total_cost += lemons_to_buy*cheap_lemon;
         cheap_lemon=lprices[i];
         lemons_to_buy=total_cups[i]*lemon_per_cup;
      }
  }
  total_cost += lemons_to_buy*cheap_lemon;